docs/maui/views/popup-service.md

```diff
@@ -133,9 +135,24 @@ The `IPopupService` also provides methods to handle a result being returned from
 
 ## Passing data to a Popup view model
 
-When presenting a Popup we sometimes need to pass data across to the underlying view model to allow for dynamic content to be presented to the user. The `IPopupService` makes this possible through the overloads of the `ShowPopup` and `ShowPopupAsync` methods that takes a `Action<TViewModel> onPresenting` parameter. This parameter has been designed to be framework agnostic and allow you as a developer to drive the loading/passing of data however best fits your architecture.
+When presenting a Popup we sometimes need to pass data across to the underlying view model to allow for dynamic content to be presented to the user. The `IPopupService` makes this possible through the overloads of the `ShowPopup` and `ShowPopupAsync` methods that takes a `IDictionary<string, object> shellParameters` parameter. This makes use of the `IQueryAttributable` interface provided with .NET MAUI Shell, for more information on using this please refer to [Process navigation data using a single method](/dotnet/maui/fundamentals/shell/navigation#process-navigation-data-using-a-single-method).
 
-To extend the previous example of showing a `NamePopupViewModel` and its associated Popup, we can use the `onPresenting` parameter to pass in the users name:
+To extend the previous example of showing a `NamePopupViewModel` and its associated Popup, we can extend the `NamePopupViewModel` to implement the `IQueryAttributable` interface as follows:
+
+```csharp
+public class NamePopupViewModel : ObservableObject, IQueryAttributable
+{
+    [ObservableProperty]
+    public partial string Name { get; set; } = "";
+
+    public void ApplyQueryAttributes(IDictionary<string, object> query)
+    {
+        Name = query["Name"] as string;
```

Comment:
I like to use `nameof` to kinda use "strongly typed" queries. Feel free to ignore this if you think it's too much and/or should not be recommended as a best practice.

Let's also use a cast (`(string)`) instead of a type  conversion (`as`) since `Name` is a non-nullable string and `as string` may return `null`.

```suggestion
        Name = (string)query[nameof(NamePopupViewModel.Name)];
```